### PR TITLE
Adding Take Screenshot Sample

### PR DESCRIPTION
**Pull request by apurvag123**

### CODE REVIEW

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            Map myMap = new Map(Basemap.CreateImagery());
+
+            // Provide used Map to the MapView
+            _myMapView.Map = myMap;
+        }
+
+        private void CreateLayout()
+        {
+            // Setup the visual frame for the MapView
+            _myMapView = new MapView()
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: Should this incorporate the top toolbar?

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            {
+                Frame = new CoreGraphics.CGRect(0, 0, View.Bounds.Width, View.Bounds.Height)
+            };
+
+            // Create a button to take the screenshot
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: Move CreateLayout to last method in class.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        private void Initialize()
+        {
+            // Create new Map with basemap
+            Map myMap = new Map(Basemap.CreateImagery());
+
+            // Provide used Map to the MapView
+            _myMapView.Map = myMap;
+        }
+
+        private void CreateLayout()
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: Move CreateLayout to last.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "Name": "Take Screenshot",
+  "SampleName": "TakeScreenshot",
+  "Description": "This sample demonstrates how you can take screenshot of a map.",
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json]: Please refer https://developers.arcgis.com/ios/beta/swift/sample-code/take-screenshot.htm for the description.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        private void OnCloseImageViewClicked(object sender, EventArgs e)
+        {
+            _overlayView.Hidden = true;
+        }
+
+        private async void OnScreenshotButtonClicked(object sender, EventArgs e)
+        {
+            // Export the image from mapview and assign it to the imagiview
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: Please fix tabs and space inside of this method.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Drawing;
+
+using CoreFoundation;
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: Please use VS refactor for usings. Reorder and remove unused.

**Comment by anttikajanus** [comment]
Please add references to the issues to this pull requests that it is resolving.

**Comment by apurvag123** [comment]
Made changes

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{
+  "Name": "Take Screenshot",
+  "SampleName": "TakeScreenshot",
+  "Description": "This sample demonstrates how you can take screenshot of a map. The app has a Screenshot button in the bottom toolbar you can tap to take screenshot of the visible area of the map. You can pan or zoom to a specific location and tap on the button, which also shows you the preview of the image produced. You can tap on the X button or any where on the preview image to close it.",
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json]: Is there X button in the view that can be pressed?

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  "Instructions": "",
+  "Type": 0,
+  "RequiresOnlineConnection":true,
+  "RequiresOfflineData":false,
+  "RequiresLocalServer":false,
+  "Image":"TakeScreenshot.jpg",
+  "Link":"",
+  "TypeLink": ["P:Esri.ArcGISRuntime.Mapping.Map.MaxScale","P:Esri.ArcGISRuntime.Mapping.Map.MinScale"
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json]: Are these correct?

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+            // Show the overlay view
+            _overlayView.Hidden = false;
+        }
+
+        private void CreateLayout()
+        {
+            // Setup the visual frame for the MapView
+            _myMapView = new MapView()
+            {
+                Frame = new CoreGraphics.CGRect(0, 0, View.Bounds.Width, View.Bounds.Height)
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: How about this?

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json (hunk 7 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json]: Fixed. Removed the X button text. And added text about the button that closes the preview

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs (hunk 9 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: I am not sure what I want to fix here.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json (hunk 8 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/metadata.json]: Fixed

**Comment by apurvag123** [comment]
Made changes. Please review.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs (hunk 9 above)
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: With this definition, is the MapView also below the toolbar or starting correctly below it?

Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: Yes it should be. All the samples use the same way of creating the map view.

Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: All the samples aren't doing that. Please investigate which approach is correct.

Here is one example of a sample that does it differently.
https://github.com/Esri/arcgis-runtime-samples-xamarin/blob/master/src/iOS/Xamarin.iOS/Samples/MapView/MapRotation/MapRotation.cs

**Comment by anttikajanus** [comment]
Please investigate the map offset issue and after that please merge.

File: src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs (hunk 9 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: I compared this sample (thats using  the offset) with mine and I don't see any difference at all. I am looking in forums on what is the right way.

Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/Map/TakeScreenshot/TakeScreenshot.cs]: I talked to the iOS team and they said that they normally use an offset of 64 points when the app has a navigation bar. So I will be using the same for our samples.

**Pull request merged**